### PR TITLE
Remove Notary docs

### PR DESCRIPTION
**Pull request by usha-mandya**
Signed-off-by: Usha Mandya <usha.mandya@docker.com>

Removed Notary docs
Added redirects from Notary docs to https://docs.docker.com/engine/security/trust/
Updated existing references to point to the upstream Notary repo https://github.com/theupdateframework/notary

### CODE REVIEW

File: engine/security/trust/index.md [hunk 1]
```diff
@@ -104,7 +113,7 @@ read how to [manage keys for DCT](trust_key_mng.md).
 
 Within the Docker CLI we can sign and push a container image with the
 `$ docker trust` command syntax. This is built on top of the Notary feature
-set, more information on Notary can be found [here](/notary/getting_started/).
+set. For more information, see the [Notary](https://github.com/theupdateframework/notary){:target="_blank" rel="noopener" class="_"} GitHub repository.
```
Comment (thaJeztah) [inline engine/security/trust/index.md:116]: Perhaps change the link caption to `Notary GitHub repository` (easier to click, and more descriptive for the link)

```suggestion
set. For more information, see the [Notary GitHub repository](https://github.com/theupdateframework/notary){:target="_blank" rel="noopener" class="_"}.
```

**Review by thaJeztah — APPROVED** [review]
left one suggestion, but LGTM otherwise

**Review by milosgajdos — DISMISSED** [review]
LGTM

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**